vpr/src/place/place_constraints.cpp

```diff
@@ -257,3 +258,149 @@ void load_cluster_constraints() {
+                num_tiles++;
+                loc.x = x;
+                loc.y = y;
+                loc.sub_tile = reg.get_sub_tile();
+                if (num_tiles > 1) {
+                    return num_tiles;
+                }
+
+                /*
+                 * If the region passed does not have a subtile set, set the
+                 * subtile to zero. The loc that is set will only be used in the
+                 * event that num_tiles is 1, and num_tiles will only be 1 if the
+                 * capacity of the tile type turns out to be zero, thus implying
+                 * that the location subtile will be zero.
+                 */
+            } else if (reg.get_sub_tile() == NO_SUBTILE) {
+                auto& cap = tile->capacity;
+                for (int z = 0; z < cap; z++) {
+                    num_tiles++;
+                    loc.x = x;
+                    loc.y = y;
+                    loc.sub_tile = 0;
+                    if (num_tiles > 1) {
+                        return num_tiles;
+                    }
+                }
+            }
+        }
+    }
+
+    return num_tiles;
+}
+
+/*
+ * Used when marking fixed blocks to check whether the ParitionRegion associated with a block
+ * covers one tile. If it covers one tile, it is marked as fixed. If it covers 0 tiles or
+ * more than one tile, it will not be marked as fixed. As soon as it is known that the
+ * PartitionRegion covers more than one tile, there is no need to check further regions
+ * and the routine will return false.
+ */
+bool is_pr_size_one(PartitionRegion& pr, t_logical_block_type_ptr block_type, t_pl_loc& loc) {
+    std::vector<Region> regions = pr.get_partition_region();
+    bool pr_size_one;
+    Region intersect_reg;
+    int pr_size = 0;
+    int reg_size;
+
+    for (unsigned int i = 0; i < regions.size(); i++) {
+        reg_size = region_tile_cover(regions[i], block_type, loc);
+
+        /*
+         * If multiple regions in the PartitionRegion all have size 1,
+         * the block may still be marked as locked, in the case that
+         * they all cover the exact same tile. To check whether this
+         * is the case, whenever there is a size 1 region, it is intersected
+         * with the previous size 1 regions to see whether it covers the same tile.
+         * If there is an intersection, it does cover the same tile, and so pr_size is
+         * not incremented (unless this is the first size 1 region encountered).
+         */
+        if (reg_size == 1) {
+            if (i == 0) {
+                intersect_reg = regions[0];
+                pr_size = pr_size + reg_size;
+            } else {
+                intersect_reg = intersection(intersect_reg, regions[i]);
+            }
+            if (intersect_reg.empty()) {
+                pr_size = pr_size + reg_size;
+                if (pr_size > 1) {
+                    break;
+                }
+            } else {
```

Comment:
This else continue clause is harder to read than it needs to be (more spaghetti code).
Instead, make the whole if (reg_size ==1) condition have an else { clause that has lines 393 - 396 in it (and then delete the continue in 388 - 390).

It may be possible to restructure the ifs more; this is a fairly complicated set of if statements for logic that isn't that fundamentally hard.